### PR TITLE
KAFKA-5091: ReassignPartitionsCommand should...

### PR DESCRIPTION
**Pull request by huxihx**
ReassignPartitionsCommand should protect against empty replica list assignment.

### CODE REVIEW

**Comment by huxihx** [comment]
@ijuma  Sorry to close the old PR due to an unrecoverable misoperation. Please review this PR. Thanks.

**Comment by huxihx** [comment]
@ewencp Could you kindly take some time to review this PR? Thanks.

**Comment by huxihx** [comment]
@hachikuji Please take some time to review this PR. Thanks.

File: core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala [hunk 1]
```diff
@@ -186,6 +186,9 @@ object ReassignPartitionsCommand extends Logging {
 
     if (partitionsToBeReassigned.isEmpty)
       throw new AdminCommandFailedException("Partition reassignment data file is empty")
+    if (partitionsToBeReassigned.exists(_._2.isEmpty)) {
+      throw new AdminCommandFailedException("Partition replica list should not be empty")
```
Comment (hachikuji) [inline core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala]: nit: "should not" -> "cannot"?

**Review by hachikuji — COMMENTED** [review]
Thanks for the patch. I think this makes sense. Can you please rename the title to "KAFKA-5091: ReassignPartitionsCommand should..."

**Comment by huxihx** [comment]
@hachikuji Thanks for the comments. Please review again.

**Review by hachikuji — APPROVED** [review]
LGTM

**Pull request closed without merging**